compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1106,6 +1106,9 @@ rustc_index::newtype_index! {
     /// are edges that go from a multi-successor node to a multi-predecessor node. This pass is
     /// needed because some analyses require that there are no critical edges in the CFG.
     ///
+    /// Note that this type is just a pointer; the actual data that a basic block holds is in
+    /// [`BasicBlockData`].
```

Comment:
nit: Consider avoiding the word "pointer" here -- maybe "index"?

Great to have the link to `BasicBlockData` in here!

Reply:
I thought of that, but it sounded kind of funny to me and I wasn't 100% sure that it was an index and not the key to a hashmap or something. I'll see if I can come up with something better...

Reply:
It's _definitely_ an index -- note that the type is defined by the `rustc_index::newtype_index!` macro.  (And the `START_BLOCK` constant is index `0`.)

Reply:
Thanks! I saw that but I think I didn't connect the dots :)